devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -591,6 +610,21 @@ private void executeIfConfigured(String pluginGroupId, String pluginArtifactId,
                         pluginManager));
     }
 
+    private String getGoalPhaseOrNull(String groupId, String artifactId, String goal, String defaultPhase) {
+        Plugin plugin = getConfiguredPluginOrNull(groupId, artifactId);
+
+        if (plugin == null) {
+            return null;
+        }
+        for (PluginExecution pluginExecution : plugin.getExecutions()) {
+            if (pluginExecution.getGoals().contains(goal)) {
+                var configuredPhase = pluginExecution.getPhase();
+                return configuredPhase != null ? configuredPhase : defaultPhase;
```

Comment:
So we are falling back to the default phase if and only if the proper goal was found but there is not explicit phase.